docs/hazmat/primitives/smime.rst

```diff
@@ -0,0 +1,125 @@
+    A1UdEwEB/wQFMAMBAf8wCgYIKoZIzj0EAwIDSQAwRgIhANES742XWm64tkGnz8Dn
+    pG6u2lHkZFQr3oaVvPcemvlbAiEA0WGGzmYx5C9UvfXIK7NEziT4pQtyESE0uRVK
+    Xw4nMqk=
+    -----END CERTIFICATE-----
+    """.strip()
+
+
+S/MIME provides a method to send and receive secure MIME messages. It is
+commonly used in email. S/MIME has multiple versions, but this
+module implements a subset of :rfc:`2632`, also known as S/MIME Version 3.
+
+
+.. class:: SMIMESignatureBuilder
+
+    .. versionadded:: 3.2
+
+    .. doctest::
+
+        >>> from cryptography.hazmat.primitives import hashes, serialization, smime
+        >>> from cryptography import x509
+        >>> cert = x509.load_pem_x509_certificate(ca_cert)
+        >>> key = serialization.load_pem_private_key(ca_key, None)
+        >>> options = [smime.SMIMEOptions.DetachedSignature]
+        >>> smime.SMIMESignatureBuilder().add_data(
+        ...     b"data to sign"
+        ... ).add_signer(
+        ...     cert, key, hashes.SHA256()
+        ... ).sign(
+        ...     serialization.Encoding.PEM, options
+        ... )
+        b'...'
+
+    .. method:: add_data(data)
+
+        :param data: The data to be hashed and signed.
+        :type data: :term:`bytes-like`
+
+    .. method:: add_signer(certificate, private_key, hash_algorithm)
+
+        :param certificate: The :class:`~cryptography.x509.Certificate`.
+
+        :param private_key: The
+            :class:`~cryptography.hazmat.primitives.asymmetric.rsa.RSAPrivateKey` or
+            :class:`~cryptography.hazmat.primitives.asymmetric.ec.EllipticCurvePrivateKey`
+            associated with the certificate provided.
+
+        :param hash_algorithm: The
+            :class:`~cryptography.hazmat.primitives.hashes.HashAlgorithm` that
+            will be used to generate the signature. This must be an instance of
+            :class:`~cryptography.hazmat.primitives.hashes.SHA1`,
+            :class:`~cryptography.hazmat.primitives.hashes.SHA224`,
+            :class:`~cryptography.hazmat.primitives.hashes.SHA256`,
+            :class:`~cryptography.hazmat.primitives.hashes.SHA384`, or
+            :class:`~cryptography.hazmat.primitives.hashes.SHA512`.
+
+    .. method:: sign(encoding, options, backend=None)
+
+        :param encoding: :attr:`~cryptography.hazmat.primitives.serialization.Encoding.PEM`
+            or :attr:`~cryptography.hazmat.primitives.serialization.Encoding.DER`.
+
+        :param options: A list of :class:`~cryptography.hazmat.primitives.smime.SMIMEOptions`.
+
+        :param backend: An optional backend.
+
+
+.. class:: SMIMEOptions
+
+    .. versionadded:: 3.2
+
+    An enumeration of options for S/MIME signature creation.
+
+    .. attribute:: Text
+
+        The text option adds ``text/plain`` headers to the S/MIME message when
+        serializing to
+        :attr:`~cryptography.hazmat.primitives.serialization.Encoding.PEM`.
+        This option has no effect with ``DER`` serialization.
```

Comment:
Shouldn't it be an error?

Reply:
Err, this was marked as resolved, but what was the resolution?

Reply:
Still pending.

Reply:
Docs need to be updated but this is disallowed on DER now. https://github.com/pyca/cryptography/pull/5465/files#diff-469789d03a0660ee2f47de1f731954feR85